src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/RSBridgePeripheral.java

```diff
@@ -0,0 +1,887 @@
+import com.refinedmods.refinedstorage.api.resource.ResourceAmount;
+import com.refinedmods.refinedstorage.common.storage.StorageTypes;
+import com.refinedmods.refinedstorage.common.support.resource.FluidResource;
+import com.refinedmods.refinedstorage.common.support.resource.ItemResource;
+import com.refinedmods.refinedstorage.mekanism.ChemicalResource;
+import com.refinedmods.refinedstorage.mekanism.ChemicalResourceType;
+import dan200.computercraft.api.lua.IArguments;
+import dan200.computercraft.api.lua.LuaException;
+import dan200.computercraft.api.lua.LuaFunction;
+import dan200.computercraft.api.lua.MethodResult;
+import dan200.computercraft.api.peripheral.IComputerAccess;
+import dan200.computercraft.core.apis.TableHelper;
+import de.srendi.advancedperipherals.common.addons.APAddons;
+import de.srendi.advancedperipherals.common.addons.computercraft.owner.BlockEntityPeripheralOwner;
+import de.srendi.advancedperipherals.common.addons.refinedstorage.RSCraftJob;
+import de.srendi.advancedperipherals.common.addons.refinedstorage.RsApi;
+import de.srendi.advancedperipherals.common.addons.refinedstorage.RsChemicalHandler;
+import de.srendi.advancedperipherals.common.addons.refinedstorage.RsFluidHandler;
+import de.srendi.advancedperipherals.common.addons.refinedstorage.RsItemHandler;
+import de.srendi.advancedperipherals.common.blocks.blockentities.RsBridgeEntity;
+import de.srendi.advancedperipherals.common.configuration.APConfig;
+import de.srendi.advancedperipherals.common.util.Pair;
+import de.srendi.advancedperipherals.common.util.inventory.ChemicalFilter;
+import de.srendi.advancedperipherals.common.util.inventory.ChemicalUtil;
+import de.srendi.advancedperipherals.common.util.inventory.FluidFilter;
+import de.srendi.advancedperipherals.common.util.inventory.FluidUtil;
+import de.srendi.advancedperipherals.common.util.inventory.GenericFilter;
+import de.srendi.advancedperipherals.common.util.inventory.IStorageSystemPeripheral;
+import de.srendi.advancedperipherals.common.util.inventory.InventoryUtil;
+import de.srendi.advancedperipherals.common.util.inventory.ItemFilter;
+import de.srendi.advancedperipherals.lib.peripherals.BasePeripheral;
+import mekanism.api.chemical.IChemicalHandler;
+import net.neoforged.neoforge.fluids.capability.IFluidHandler;
+import net.neoforged.neoforge.items.IItemHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+public class RSBridgePeripheral extends BasePeripheral<BlockEntityPeripheralOwner<RsBridgeEntity>> implements IStorageSystemPeripheral {
+
+    public static final String PERIPHERAL_TYPE = "rsBridge";
+
+    private final RsBridgeEntity bridge;
+
+    public RSBridgePeripheral(RsBridgeEntity owner) {
+        super(PERIPHERAL_TYPE, new BlockEntityPeripheralOwner<>(owner));
+        this.bridge = owner;
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
```

Comment:
does config reading have to run on main thread?
or I think `isEnabled` usually is not a Lua API here?

Reply:
Shouldn't be a lua function, not sure why I annoted it